Tests/Get-CEventLog.Tests.ps1

```diff
@@ -0,0 +1,127 @@
+
+#Requires -RunAsAdministrator
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+BeforeAll {
+    Set-StrictMode -Version 'Latest'
+
+    & (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Test.ps1' -Resolve)
+
+    function GivenEventLog
+    {
+        New-CEventLog -LogName $script:logName -Source 'Get-CEventLogs.Tests'
+    }
+
+    function GivenLogEntry
+    {
+        param(
+            [String] $Message,
+            [Diagnostics.EventLogEntryType] $EntryType = 'Error'
+        )
+        [Diagnostics.EventLog]::WriteEntry($script:logName, $Message, $EntryType)
+    }
+
+    function WhenGettingEventLogs
+    {
+        param(
+            [switch] $List,
+            [String] $LogName,
+            [int] $Newest,
+            [Diagnostics.EventLogEntryType] $EntryType,
+            [String] $Message
+        )
+
+        $script:result = Get-CEventLog @PSBoundParameters
+    }
+
+    function ThenEventLogsAreListed
+    {
+        $script:result | Should -BeOfType 'System.Diagnostics.EventLog'
+        $script:result | Should -Not -BeNullOrEmpty
+    }
+
+    function CleanUpEventLog
+    {
+        [Diagnostics.EventLog]::Delete($script:logName)
+    }
+}
+
+Describe 'Get-CEventLog' {
+    BeforeEach {
+        $script:result = $null
+        $script:logName = 'Get-CEventLogs.Tests'
+        Start-Sleep -Seconds 1
+    }
+
+    AfterEach {
+        Uninstall-CEventLog -LogName $script:logName
+    }
+
+    It 'should list event logs' {
+        WhenGettingEventLogs -List
+        ThenEventLogsAreListed
+    }
+
+    It 'should list event logs including the defaults' {
+        GivenEventLog

```

Comment:
I want to use a global variable so that the `AfterEach` block is able to refer to the `LogName` when uninstalling.

Could I create a `GivenLogName` function that takes in a LogName and assigns it to `$script:logName`?

Reply:
I have the same question for the other test files as well.

Reply:
Save the value you want to cleanup in a variable that gets set in `When` and used in `AfterEach` .